### ISSUE TITLE
calling save on a resource should add itself to its collection

### ISSUE THREAD

**Issue by wes-r** [masylum/mobx-rest#51]
`Model.save` on a new model should add itself to its collection (if it has one). This was the behavior I expected, especially after playing with `destroy` and how it correctly removes itself from its collection.

Simple example:
```
let newModel = collection.build();
await newModel.save();
collection.get(newModel.id) == newModel; // This is what is expected!
```

**Comment by andresgutgon** [comment]
You could use [collection.create](https://github.com/masylum/mobx-rest/blob/master/src/Collection.js#L242)
``` javascript
collection.create({ name: 'Jon', last_name: 'Doe' })
```

**Comment by wes-r** [comment]
True, but we often deal with models that are either new OR existing (new/editing). It'd be nice to just call `model.save()` and have it add itself to the collection, instead of needing to do:

```
model.isNew ? model.collection.create(model) : model.save();
```

**Comment by masylum** [comment]
This makes a lot of sense!

**Comment by masylum** [comment]
https://github.com/masylum/mobx-rest/pull/79

**Comment by jeacott1** [comment]
this appears to be a dupe and should be closed.
